wordpress-mobile/WordPress-Android
# Dashboard Cards: Add Activity Log to card source

Closes #18207 

This PR
- Adds logic to fetch Activity cards from CardsSource implemented in this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2712)
- Adds unit tests to CardsSourceTest

There are no visual UI changes within this PR and the app should function as normal.

**Merge Instructions**
- Make sure the Dependent FluxC PR has been merged into trunk & CI Publish task to complete
- Update the FluxC PR hash version within build.gradle 
- Wait for CI to complete after the push
- Remove the Not Ready for Merge Label
- Merge as normal
**OR**
- Approve the changes and leave the merging for me


**To Test**
**Note**: This test requires you to use the Android Studio: App Inspection > Database Inspector to verify records are retreived
- Install the Jetpack app (or download the branch and run using AS)
- Login with a wp account
- Navigate to the dashboard
- Connect your device to Android Studio
- Navigate to App Inspection
- Select the device that is running the installed JP version
- Navigate to Database Inspector
- Locate the `wp-android-database` and select "DashboardCards"
- ✅ Verify that the database does not include an "ACTITIITY" type row
------------------
- Navigate to App Settings > Debug Settings
- Enable feature flag `dashboard_card_activity_log` and restart the app
- Navigate to the dashboard
- Navigate to App Inspection
- You will need to re-select the device that is running the installed JP version
- Navigate to Database Inspector
- Locate the `wp-android-database` and select `DashboardCards`
- ✅ Verify that the database does include an "**ACTIVITY**" type row

## Regression Notes
1. Potential unintended areas of impact
The dashboard cards are not showing properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit and manual testing

3. What automated tests I added (or what prevented me from doing so)
Add tests to `CardSourceTest`

PR submission checklist:
